include/zephyr/bluetooth/buf.h

```diff
@@ -161,6 +161,8 @@ static inline enum bt_buf_type bt_buf_get_type(struct net_buf *buf)
 		->type;
 }
 
+struct net_buf *bt_buf_get_evt_but_better(uint8_t evt, uint8_t meta, k_timeout_t timeout);
```

Comment:
I suppose we'd want to deprecate `bt_buf_get_evt()` as well? That said, you'd need to update all upstream HCI drivers first, I suppose.

Reply:
yes. I made this to allow a soft-migration. Now that I think of it, I should update `bt_buf_can_steal_ext_adv_report_buf` to take `buf` and double-check the pool too. Because it won't work if the driver uses a different pool.

Reply:
fixed. Found (and fixed) a bug in the process, yay.